python/shotgun_desktop/startup.py

```diff
@@ -478,237 +334,118 @@ def __extract_command_line_argument(arg_name):
     return is_set
 
 
-def __launch_app(app, splash, connection, app_bootstrap, server, settings):
+def __launch_app(app, splash, user, app_bootstrap, server, settings):
     """
     Shows the splash screen, optionally downloads and configures Toolkit, imports it, optionally
     updates it and then launches the desktop engine.
 
     :param app: Application object for event processing.
     :param splash: Splash dialog to update user on what is currently going on
-    :param connection: Connection to the Shotgun server.
+    :param user: Current ShotgunUser.
+    :param app_bootstrap: Application bootstrap.
     :param server: The tk_framework_desktopserver.Server instance.
     :param settings: The application's settings.
 
     :returns: The error code to return to the shell.
     """
     # show the splash screen
     splash.show()
+
+    import sgtk
+
+    sgtk.set_authenticated_user(user)
+
+    # Downloads an upgrade for the startup if available. The startup upgrade is independent from the
+    # auto_path state and has its own logic for auto-updating or not, so move this outside the
+    # if auto_path test.
+    startup_updated = upgrade_startup(
+        splash,
+        sgtk,
+        app_bootstrap
+    )
+    if startup_updated:
+        __restart_app_with_countdown(splash, "Shotgun Desktop updated.")
+
     splash.set_message("Looking up site configuration.")
     app.processEvents()
 
+    connection = user.create_sg_connection()
+
     _assert_toolkit_enabled(splash, connection)
 
-    logger.debug("Getting the default site config")
-    default_site_config, pc = shotgun_desktop.paths.get_default_site_config_root(connection)
+    logger.debug("Getting the default site configuration.")
+    pc_path, pc, toolkit_classic_required = shotgun_desktop.paths.get_default_site_config_root(connection)
```

Comment:
ahh yes, maybe all that's needed is renaming this method to `get_pipeline_confiugration_info` or something like that! :)